sled-agent/src/sim/artifact_store.rs

```diff
@@ -83,6 +105,12 @@ impl ArtifactStore<SimArtifactStorage> {
             .await
             .unwrap();
     }
+
+    pub fn create_delete_watcher(&self) -> watch::Receiver<Generation> {
+        let mut watcher = self.storage.delete_done_rx.clone();
+        watcher.mark_unchanged();
+        watcher
```

Comment:
Nit - I think we could return `self.storage.delete_done_tx.subscribe()` instead? Then we wouldn't need the `mark_unchanged()`, and maybe could also drop the `delete_done_rx` field entirely?

Reply:
Oh, yes, we could do that. I forgot about the `subscribe()` method.